### PR TITLE
add GAMMA dockerfile

### PR DESCRIPTION
**Pull request by stjacqrm**

### CODE REVIEW

File: gamma/1.4/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# base image
+FROM ubuntu:bionic
+
+# metadata
+LABEL base.image="ubuntu:bionic"
+LABEL container.version="1"
```
Comment (kapsakcj) [inline gamma/1.4/Dockerfile:6]: please change `LABEL container.version="1"` to `LABEL dockerfile.version="1"`

File: gamma/1.4/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+LABEL software="centroid"
+LABEL software.version="v1.4"
+LABEL description="Gene Allele Mutation Microbial Assessment"
+LABEL website="https://github.com/rastanton/GAMMA/tree/v1.4"
+LABEL license=""
```
Comment (kapsakcj) [inline gamma/1.4/Dockerfile:11]: Maybe put none for license? GAMMA GitHub repo doesn't show a license but the [bioconda page says Apache 2.0](https://bioconda.github.io/recipes/gamma/README.html) 

We could ask Rich to add a license to the GH repo

File: gamma/1.4/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    curl \
+    make \
+    g++ \
+    ncbi-blast+ \
+    hmmer && \
+    apt-get autoclean && \
+    rm -rf /var/lib/apt/lists/*
+
+#install biopython
+RUN pip3 install biopython cgecore
```
Comment (kapsakcj) [inline gamma/1.4/Dockerfile:33]: I would recommend combining all of these `pip3 install` layers into one, unless there is some reason not to that I'm unaware of.

`RUN pip3 install biopython cgecore tabulate` should install all at once.

**Comment by kapsakcj** [comment]
OK I pushed a commit with the changes I suggested. Dockerfile builds successfully and GAMMA runs successfully (with a BioPython warning) with the included ResFinder database:
```
$ GAMMA.py GCA_011245895.1_PDT000672941.1_genomic.fna /GAMMA/ResFinderDB_Combined_05-06-20.fsa 2021-12-29-gamma-testing/output.gamma.tsv
Loaded 2833474 letters in 3087 sequences
Searched 4922072 bases in 67 sequences
/usr/local/lib/python3.6/dist-packages/Bio/Seq.py:2983: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.
  BiopythonWarning,
```
Also ran successfully with the NCBI-amrfinderplus database:
```
$ GAMMA.py GCA_011245895.1_PDT000672941.1_genomic.fna /amrfinder/data/2021-12-21.1/AMR_CDS outputs/2021-12-29-gamma-testing/output.gamma.amrfinderDB.tsv
Loaded 7059547 letters in 7024 sequences
Searched 4922072 bases in 67 sequences
/usr/local/lib/python3.6/dist-packages/Bio/Seq.py:2983: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.
  BiopythonWarning,
```

But it failed to run with the ARG-ANNOT database. It complains about duplicate entries:
```
$ GAMMA.py GCA_011245895.1_PDT000672941.1_genomic.fna /arg-annot.fasta 2021-12-29-gamma-testing/output.gamma.tsv
Loaded 1818898 letters in 2038 sequences
Error: sequence name (Tet)Tet(48):NG_055773:1-1200:1200 is repeated in the database, all names must be unique.
Traceback (most recent call last):
  File "/GAMMA/GAMMA.py", line 1033, in <module>
    sys.exit(main())
  File "/GAMMA/GAMMA.py", line 1026, in main
    GAMMA_Out(args.input_fasta, args.database, args.output,  args.percent_identity, Verbose)
  File "/GAMMA/GAMMA.py", line 1001, in GAMMA_Out
    GAMA_Output(output +'.psl', fasta, gene_db, output + '.gamma', verbose)
  File "/GAMMA/GAMMA.py", line 929, in GAMA_Output
    List1 = GAMA_List(PSL, genome_fasta, genes_fasta, verbose)
  File "/GAMMA/GAMMA.py", line 919, in GAMA_List
    Lines = GAMA_Line_Maker(PSL, genome_fasta, genes_fasta, verbose)
  File "/GAMMA/GAMMA.py", line 728, in GAMA_Line_Maker
    Genes = SeqIO.to_dict(SeqIO.parse(genes_fasta, 'fasta'))
  File "/usr/local/lib/python3.6/dist-packages/Bio/SeqIO/__init__.py", line 737, in to_dict
    raise ValueError("Duplicate key '%s'" % key)
ValueError: Duplicate key '(Tet)Tet(48):NG_055773:1-1200:1200'
```
^ This will require removing duplicates from the ARG-ANNOT database or finding a copy that has had duplicates removed.

I'm going to go ahead and merge this PR & build on dockerhub & quay, but be aware that the included ARG-ANNOT database will cause an error

**Comment by kevinlibuit** [comment]
Initially thought the ARG-ANNOT failure was due to use of [the nucleotide database rather](https://github.com/StaPH-B/docker-builds/blob/master/gamma/1.4/Dockerfile#L58) than the amino acid one required for GAMMA, but it seems that [GAMMA still fails due to duplicate entries in the AA database cited in the GAMMA README](https://github.com/rastanton/GAMMA/issues/6)

**Comment by kapsakcj** [comment]
FYI Rich had the wrong link in the GAMMA README for the arg-annot database. I updated the dockerfile with the correct link and now GAMMA successfully runs on the arg-annot database 👍 Rebuilding on dockerhub & quay now

commit: https://github.com/StaPH-B/docker-builds/commit/2da0aa0f3d8b8adb7268cabf39b7a9e89a811066

**Pull request merged**